src/main/java/com/autotune/analyzer/recommendations/updater/vpa/VpaUpdaterImpl.java

```diff
@@ -0,0 +1,371 @@
+
+                // create VPA Object is not present
+                if (!vpaPresent) {
+                    createVpaObject(kruizeObject);
+                }
+
+                for (K8sObject k8sObject: kruizeObject.getKubernetes_objects()) {
+                    List<RecommendedContainerResources> containerRecommendations = convertRecommendationsToContainerPolicy(k8sObject.getContainerDataMap());
+                    if (containerRecommendations.isEmpty()){
+                        LOGGER.error(AnalyzerErrorConstants.RecommendationUpdaterErrors.RECOMMENDATION_DATA_NOT_PRESENT);
+                    } else {
+                        RecommendedPodResources recommendedPodResources = new RecommendedPodResources();
+                        recommendedPodResources.setContainerRecommendations(containerRecommendations);
+                        VerticalPodAutoscalerStatus vpaObjectStatus = new VerticalPodAutoscalerStatusBuilder()
+                                .withRecommendation(recommendedPodResources)
+                                .build();
+
+                        // patching existing VPA Object
+                        if (vpaObjectStatus != null) {
+                            VerticalPodAutoscaler vpaObject = getVpaIsPresent(expName);
+                            vpaObject.setStatus(vpaObjectStatus);
+
+                            NamespacedVerticalPodAutoscalerClient client = new DefaultVerticalPodAutoscalerClient();
+                            client.v1().verticalpodautoscalers()
+                                    .inNamespace(vpaObject
+                                            .getMetadata()
+                                            .getNamespace())
+                                    .withName(vpaObject
+                                            .getMetadata()
+                                            .getName())
+                                    .patchStatus(vpaObject);
+
+                            LOGGER.info(String.format(AnalyzerConstants.RecommendationUpdaterConstants.InfoMsgs.VPA_PATCHED,
+                                    vpaObject.getMetadata().getName()));
+                        }
+                    }
+                }
+
+            } else {
+                LOGGER.error(AnalyzerErrorConstants.RecommendationUpdaterErrors.UPDATER_NOT_INSTALLED);
+            }
+        } catch (Exception e) {
+            throw new ApplyRecommendationsError(e.getMessage());
+        }
+    }
+
+    /**
+     * This function converts container recommendations for VPA Container Recommendations Object Format
+     */
+    private List<RecommendedContainerResources>  convertRecommendationsToContainerPolicy(HashMap<String, ContainerData> containerDataMap) {
+        List<RecommendedContainerResources> containerRecommendations = new ArrayList<>();
+
+        for (Map.Entry<String, ContainerData> containerDataEntry : containerDataMap.entrySet()) {
+            // fetcing container data
```

Comment:
Typo in the word `fetching`

Reply:
Oh, I missed that! I’ve corrected the typo now.